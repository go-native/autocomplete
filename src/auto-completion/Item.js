import React, { Component } from 'react';

class Item extends Component {
  constructor(props) {
    super(props);
    this.handleHover = this.handleHover.bind(this);
  }
  handleHover() {
    this.props.onActive();
  }
  render() {
    const itemClass = this.props.active ? "list-group-item active" : "list-group-item";
    return (<li
      onMouseEnter={this.handleHover}
      onMouseLeave={this.handleHover}
      onMouseDown={() => this.props.onSelect()}
      className={itemClass}>{this.props.caption}
    </li>)
  }
}

export default Item;
