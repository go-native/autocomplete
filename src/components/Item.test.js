import React from 'react';
import renderer from 'react-test-renderer';
import Item from './Item';

describe('<AutoCompletion />', () => {
  it('renders', () => {
    const tree = renderer
      .create(
        <Item key={1} onSelect={() => {}} caption="Hey" active={false} onActive={() => {}} />,
    )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
