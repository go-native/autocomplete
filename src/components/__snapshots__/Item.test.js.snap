// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AutoCompletion /> renders 1`] = `
<li
  className="list-group-item"
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Hey
</li>
`;
